File: backend/app/api/routes/analytics.py
from fastapi import APIRouter, Header

from ...schemas import ChatRequest, ChatResponse
from ...services.analytics import generate_response
from ...socketio_app import emit_session_event

router = APIRouter()


@router.post("/analytics-chat", response_model=ChatResponse)
async def analytics_chat(
    payload: ChatRequest,
    authorization: str | None = Header(default=None),
) -> ChatResponse:
    context = dict(payload.context or {})
    if payload.sessionId:
        context.setdefault("sessionId", payload.sessionId)
    if payload.language:
        context.setdefault("language", payload.language)
    if authorization and authorization.lower().startswith("bearer "):
        token = authorization.split(" ", 1)[1]
        context["authToken"] = token
    elif authorization:
        context["authToken"] = authorization

    result = await generate_response(payload.persona, payload.prompt, context)
    metadata = result.get("metadata") if isinstance(result, dict) else None
    session_id = metadata.get("sessionId") if isinstance(metadata, dict) else None
    if session_id:
        await emit_session_event(
            session_id,
            "analytics:response",
            {
                "persona": result.get("persona"),
                "messages": result.get("messages"),
                "metadata": metadata,
            },
        )
    return ChatResponse.model_validate(result)

File: frontend/src/api/analytics.ts
import { apiClient } from "./client";

export type PersonaKey = "priya" | "jax" | "lena" | "mike" | "amara" | "kabir" | string;

export type ChatAttachment = {
  type: string;
  title?: string;
  media?: string;
  url?: string;
  cta?: string;
  payload?: unknown;
};

export type ChatMessage = {
  role: string;
  content: string;
  attachments?: ChatAttachment[];
};

export type ChatResponse = {
  persona: PersonaKey;
  messages: ChatMessage[];
  metadata?: {
    sessionId?: string;
    user?: Record<string, unknown> | null;
    maas?: Record<string, unknown>;
  };
};

export type ChatRequest = {
  persona: PersonaKey;
  prompt: string;
  sessionId?: string;
  language?: string;
  context?: Record<string, unknown>;
};

export async function sendAnalyticsPrompt(
  payload: ChatRequest,
  token?: string,
): Promise<ChatResponse> {
  const headers: Record<string, string> = {};
  if (token) {
    headers.Authorization = token.startsWith("Bearer ") ? token : `Bearer ${token}`;
  }

  const response = await apiClient.post<ChatResponse>("/analytics-chat", payload, { headers });
  return response.data;
}

